packages/flutter_tools/lib/src/commands/migrate.dart

```diff
@@ -0,0 +1,84 @@
+import '../base/logger.dart';
+// import '../base/platform.dart';
+import '../base/terminal.dart';
+import '../migrate/migrate_utils.dart';
+import '../runner/flutter_command.dart';
+// TODO(garyq): Add each of these back in as they land.
+// import 'migrate_abandon.dart';
+// import 'migrate_apply.dart';
+// import 'migrate_resolve_conflicts.dart';
+// import 'migrate_start.dart';
+// import 'migrate_status.dart';
+
+/// Base command for the migration tool.
+class MigrateCommand extends FlutterCommand {
+  MigrateCommand({
+    // bool verbose = false,
+    required this.logger,
+    // TODO(garyq): Add each of these back in as they land.
+    // required FileSystem fileSystem,
+    // required Terminal terminal,
+    // required Platform platform,
+    // required ProcessManager processManager,
+  }) {
+    // TODO(garyq): Add each of these back in as they land.
+    // addSubcommand(MigrateAbandonCommand(logger: logger, fileSystem: fileSystem, terminal: terminal, platform: platform, processManager: processManager));
+    // addSubcommand(MigrateApplyCommand(verbose: verbose, logger: logger, fileSystem: fileSystem, terminal: terminal, platform: platform, processManager: processManager));
+    // addSubcommand(MigrateResolveConflictsCommand(logger: logger, fileSystem: fileSystem, terminal: terminal));
+    // addSubcommand(MigrateStartCommand(verbose: verbose, logger: logger, fileSystem: fileSystem, platform: platform, processManager: processManager));
+    // addSubcommand(MigrateStatusCommand(verbose: verbose, logger: logger, fileSystem: fileSystem, platform: platform, processManager: processManager));
+  }
+
+  final Logger logger;
+
+  @override
+  final String name = 'migrate';
+
+  @override
+  final String description = 'Migrates flutter generated project files to the current flutter version';
+
+  @override
+  String get category => FlutterCommandCategory.project;
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{};
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    return const FlutterCommandResult(ExitStatus.fail);
+  }
+}
+
+Future<bool> gitRepoExists(String projectDirectory, Logger logger, MigrateUtils migrateUtils) async {
+  if (await migrateUtils.isGitRepo(projectDirectory)) {
+    return true;
+  }
+  logger.printStatus('Project is not a git repo. Please initialize a git repo and try again.');
+  printCommandText('git init', logger);
+  return false;
+}
+
+Future<bool> hasUncommittedChanges(String projectDirectory, Logger logger, MigrateUtils migrateUtils) async {
+  if (await migrateUtils.hasUncommittedChanges(projectDirectory)) {
+    logger.printStatus('There are uncommitted changes in your project. Please git commit, abandon, or stash your changes before trying again.');
+    return true;
+  }
+  return false;
+}
+
+/// Prints a command to logger with appropriate formatting.
+void printCommandText(String command, Logger logger) {
```

Comment:
This is only called once, above in `gitRepoExists`. It could be a library private function.

Reply:
This will soon be used in all of the subcommands